src/components/Articles/Article/Comments/CommentCard/CommentCard.jsx

```diff
@@ -1,7 +1,43 @@
-import { item, content, votes, author, body } from "./CommentCard.module.css";
+import { item, content, votes, author, body, del } from "./CommentCard.module.css";
+import DeleteIcon from "@mui/icons-material/Delete";
+import { deleteComment } from "../../../../../api";
+import { useContext } from "react";
+import { UserContext } from "../../../../Context/UserProvider";
 
-const CommentCard = ({ comment }) => {
-  const date = new Date(comment.created_at);
+const CommentCard = ({ comments, comment, setComments, setCommentCount, setShowComments }) => {
+  const { user } = useContext(UserContext);
+
+  const handleClick = async () => {
+    const backup = { ...comment };
+    try {
+      setComments((currComments) => {
+        const filteredComments = currComments.filter((item) => {
+          return item.comment_id !== comment.comment_id;
```

Comment:
if you have an optimistically rendered comment that you can't delete, because it has no id, and doesn't technically exist in the db until refresh, you could give it an extra unique id, e.g a negative number, and add it to your filter condition as well so that it is removed as well? 
or you can only set the comment with the comment from the API (ie. not optimistically render the comment) because it's fast enough i reckon to get the posted comment back from the API.  therefore you'll be dealing with a comment that exists?

Reply:
Ahhh that makes sense - thanks for your awesome feedback as always!